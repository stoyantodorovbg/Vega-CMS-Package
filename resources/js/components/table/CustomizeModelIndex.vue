<template>
    <div class="col-lg-12">
        <button @click="changeEditing" type="button" class="btn btn-main">{{ getButtonValue() }}</button>
        <div v-if="editing" class="form-group custom-grid-panel">
            <div class="form-check"
                 v-for="field in fields"
                 :key="field.name">

                <b-form-checkbox
                        class="form-check-input"
                        :name="field.name"
                        :checked="field.visibility"
                        @change="checkVisibility(field.name, field.visibility, field.position)">

                        <div class="slot">
                            {{ field.name }}
                        </div>
                </b-form-checkbox>
            </div>
        </div>
    </div>
</template>

<script>
    export default {
        name: 'customize-model-index',

        props: {
            fields: {
                type: Array,
                default: function () {
                    return [
                        {
                            name: 'id',
                            position: 0,
                            visibility: true,
                        }
                    ];
                }
            }
        },

        data() {
            return {
                editing: false,
            }
        },

        methods: {
            checkVisibility(fieldName, fieldVisibility, position) {
                fieldVisibility = !fieldVisibility;
                this.$parent.changeFieldVisibility(fieldName, fieldVisibility, position);
            },
            getButtonValue() {
                return this.editing ? 'Customize grid - hide menu' : 'Customize grid - show menu';
            },
            changeEditing() {
                this.editing = !this.editing;
            }
        },
        mounted() {
          // console.log(this.fields);
        }
    }
</script>
