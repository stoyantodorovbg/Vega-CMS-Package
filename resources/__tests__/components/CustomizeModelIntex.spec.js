import { mount } from '@vue/test-utils'
import CustomizeModelIndex from "../../js/components/table/CustomizeModelIndex";

describe('CustomizeModelIndex Component', () => {
  test('should mount without crashing', () => {
    const wrapper = mount(CustomizeModelIndex);

    expect(wrapper).toMatchSnapshot();
  });
});
