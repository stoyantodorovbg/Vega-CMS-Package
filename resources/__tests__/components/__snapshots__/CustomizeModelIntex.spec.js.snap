// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomizeModelIndex Component should mount without crashing 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
